### ISSUE TITLE
[KratosCore] Error converting sliced numpy array to Kratos.Matrix

### ISSUE THREAD

**Issue by Rbravo555** [KratosMultiphysics/Kratos#11610]
**Description**
We have spotted a bug when converting a numpy array to a Kratos.Matrix

If the numpy array is sliced before the convertion, the Kratos.Matrix will not contain the expected entries of the Numpy Array, althogh it will have the correct shape.

**Scope**
- KratosCore

**To Reproduce**

An example code is here:

```python
import numpy as np
import KratosMultiphysics

increasing_column_matrix = np.arange(0,5,1).astype(np.float64)
increasing_column_matrix = np.repeat(increasing_column_matrix.reshape(1,-1),5, axis=0)

columns_to_keep = 3

#convert to Kratos
KratosArray = KratosMultiphysics.Matrix(increasing_column_matrix[:,:columns_to_keep])

print('complete numpy array: \n', increasing_column_matrix)
print(f'numpy array truncated to {columns_to_keep} columns: \n', increasing_column_matrix[:,:columns_to_keep])
print(f'Kratos matrix truncated to {columns_to_keep} columns: \n', KratosArray)
```

The outputs look like this:

```

complete numpy array: 
 [[0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]]
numpy array truncated to 3 columns: 
 [[0. 1. 2.]
 [0. 1. 2.]
 [0. 1. 2.]
 [0. 1. 2.]
 [0. 1. 2.]]
Kratos matrix truncated to 3 columns: 
 [5,3]((0,1,2),(3,4,0),(1,2,3),(4,0,1),(2,3,4))
```

**Expected behavior**
As can bee seen, the slicing is respected only for the shape of the Kratos.Matrix, the entries however are added to the Kratos.Matrix in consecutive order. 

**Environment**

- OS: Ubuntu 20.04.4 LTS
- Branch: master
- Python 3.8.10

**Comment by loumalouomega** [comment]
Huuum, maybe we should implement a custom constructor for numpy arrays instead of whatever we are doing right now

**Comment by loumalouomega** [comment]
Pybind11 supports it

**Comment by matekelemen** [comment]
I imagine numpy doesn't allocate new memory for a sliced `numpy.ndarray`, but defines new strides and use the original pointer with an offset. However, the `Kratos::Matrix` bindings don't consider that.

https://github.com/KratosMultiphysics/Kratos/blob/b1a2b325523bcec5f168d5e3763852a06246fe75/kratos/python/add_matrix_to_python.cpp#L79-L85

The flattened index `count` should should respect `info.strides`.

**Comment by Rbravo555** [comment]
fixed by #11627
